Create stash.md for Git Stash

reganmarie opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

- Closes issue #3489 
- Created stash.md on filepath docs/content/git/concepts/stash/stash.md
- Added Title, Description, Subjects, Tags, and Catalog Content
- Added description, syntax, and example
- Ran through three online plagiarism checkers to ensure 100% unique wording and description, passed with 100% on all three
- Followed content and style guidelines
- Checked for spelling and grammar errors
- Checked for clarity and readability

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

caupolicandiaz commented on content/git/concepts/stash/stash.md:
```diff
@@ -0,0 +1,80 @@
+---
+Title: 'Stash'
+Description: 'Git stash is used to save changes locally so the developer can switch branches while developing or to get a clean version of the branch they are currently working on.'
```
```suggestion
Description: 'A command for saving changes locally so a developer can switch branches in the midst of changes or to get a clean version of the branch they are currently working on.'
```

caupolicandiaz commented on content/git/concepts/stash/stash.md:
```diff
@@ -0,0 +1,80 @@
+  - 'learn-git'
+---
+
+Stash is used when a developer would like to save changes locally to their machine rather than committing them. This is often used when a developer would like to switch branches in the middle of coding but the code they are working on is not yet ready to be committed.
```
```suggestion
The **`stash`** command is used when a developer would like to save changes locally to their machine rather than committing them. This is often used when a developer would like to switch branches in the middle of an effort but the code they are working on is not yet ready to be committed.
```

caupolicandiaz commented on content/git/concepts/stash/stash.md:
```diff
@@ -0,0 +1,80 @@
+To re-apply the changes that were stashed, the developer has two options:
+
+- `git stash pop` is used in the first example and takes the changes from the stash to re-apply them.
+- `git stash apply` is used in the second example and re-applies the changes while also keeping them in the stash.
```
```suggestion
- `git stash apply` is used in the second example and reapplies the changes while also keeping them in the stash.
```

caupolicandiaz commented on content/git/concepts/stash/stash.md:
````diff
@@ -0,0 +1,80 @@
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+There are eight additional options for ‘git stash`:
````
```suggestion
There are eight additional options for `git stash`:
```

caupolicandiaz reviewed (changes requested):
@reganmarie I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

reganmarie commented:
@caupolicandiaz Updated with the suggested changes:
- Updated description in line 3 to: 'A command for saving changes locally so a developer can switch branches in the midst of changes or to get a clean version of the branch they are currently working on.'
- Updated in-depth description in line 15 to include The **`stash`** command  and changes coding to effort
- Updated re-apply or re-applies to be reapply and reapplies in line 45
- Updated ' to ` in line 70

caupolicandiaz reviewed (approved):
@reganmarie Looks good, this is ready for a second review.

reganmarie commented:
Thank you @caupolicandiaz! ☺️
